website/docs/cli/commands/test.mdx

```diff
@@ -46,9 +46,9 @@ Each Terraform test file will maintain all Terraform state it requires within me
 
 ### Terraform Test Cleanup
 
-The Terraform `terraform test` command creates _real_ infrastructure. Once Terraform fully executes each test file, Terraform attempts to destroy any remaining infrastructure. If it cannot do this, Terraform reports a list of resources it created but could not destroy.
+The Terraform `terraform test` command creates _real_ infrastructure. Once Terraform fully executes each test file, Terraform attempts to destroy any remaining infrastructure. If it cannot do this, Terraform writes the leftover state to the local file system and reports the resources it created but could not destroy.
 
-You should monitor the output of the test command closely to ensure Terraform removes the infrastructure it created or perform manual cleanup if not. We recommend creating dedicated testing accounts within the target providers that you can routinely and safely purge to ensure any accidental and costly resources aren't left behind.
+Subsequent `terraform test` commands will not be able to execute until the leftover state is removed. You can then remove the leftover state by running `terraform test cleanup` or manually removing the state files.
```

Comment:
It should be obvious to users, but in this section it may be helpful to call out that the user will need to manually address cleanup of the remote objects that are described in the state. If a user deletes the state files then they lose information about the resources/remote objects that they need to look for.